packages/types/src/DidDetails.ts

```diff
@@ -218,3 +164,15 @@ export type DidSignature = {
 export type TypedValue<K extends string, V> = K extends any
   ? { [P in K]: V }
   : never
+
+export interface DidDetails {
+  identifier: DidIdentifier
+  uri: DidUri
+
+  authentication: [DidVerificationKey]
```

Comment:
Why are these optional? Since they are array, I think it would be easier to make them empty instead of undefined.

Reply:
What is the correct way according to spec @ntn-x2 ? It seems like the current did resolver does leave empty things out...

Reply:
True, but that is logic that could be left to the resolver when showing a DID Document for a given DID. At the SDK level, at least in my opinion, it is confusing to have an optional array property, where you would have to check both for existence AND length.

Reply:
For us the developers of the SDK this might be slightly inconvenient. For the users of the SDK it would be quite confusing and inconvenient to always have some unused empty arrays hanging around. It would be even worse to have to explicitly pass them somewhere. I feel that "the required array can be empty" is a poor API.